README-template.md

```diff
@@ -426,11 +435,165 @@ Please note there are a few caveats:
+3. Create an external table definition in BigQuery that points to the data file URIs in GCS.
+4. Create an external table definition in Hive that points to the BigQuery table definition.
+
+### Example: CSV table
+
+In this example, we create simple table backed by some CSV files.
+
+The files have the following layout in a GCS bucket:
+
+```text
+gs://mybucket/warehouse/my_csv_table/file1.csv
+gs://mybucket/warehouse/my_csv_table/file2.csv
+```
+
+External BigQuery table definition:
+
+```sql
+CREATE EXTERNAL TABLE `myproject.mydataset.my_csv_table`
+WITH CONNECTION `myproject.us.myconnection`
+OPTIONS (
+  format = 'CSV',
+  uris = ['gs://mybucket/warehouse/my_csv_table/*.csv']
+)
+```
+
+External Hive table definition:
+
+```sql
+CREATE EXTERNAL TABLE mydb.my_csv_table (
+    abc BIGINT,
+    xyz STRING
+)
+STORED BY 'com.google.cloud.hive.bigquery.connector.BigQueryStorageHandler'
+TBLPROPERTIES (
+  'bq.table' = 'myproject.mydataset.my_csv_table'
+)
+```
+
+Sample Hive query to read data:
+
+```sql
+SELECT abc
+FROM mydb.my_csv_table
+WHERE xyz = "some text"
+```
+
+### Example: Partitioned Parquet table
+
+In this example, we integrate a partitioned Parquet table with BigLake and Hive.
+
+The Parquet files have two columns:
+* `id`: 64-bit integer
+* `name`: string
+
+The Parquet files are partitioned by `city` and have the following layout:
+
+```text
+gs://mybucket/warehouse/my_parquet_table/city=Paris/000000_0.parquet
+gs://mybucket/warehouse/my_parquet_table/city=Paris/000001_0.parquet
+
+gs://mybucket/warehouse/my_parquet_table/city=London/000000_0.parquet
+gs://mybucket/warehouse/my_parquet_table/city=London/000001_0.parquet
+gs://mybucket/warehouse/my_parquet_table/city=London/000002_0.parquet
+```
+
+External BigQuery table definition:
+
+```sql
+CREATE EXTERNAL TABLE `myproject.mydataset.my_parquet_table`
+WITH PARTITION COLUMNS
+WITH CONNECTION `myproject.us.myconnection`
+OPTIONS (
+  format = 'PARQUET',
+  uris = ['gs://mybucket/warehouse/my_parquet_table/*'],
+  hive_partition_uri_prefix = 'gs://mybucket/warehouse/my_parquet_table',
+  require_hive_partition_filter = true
+)
+```
+
+Notes about the above statement:
+* The `WITH PARTITION COLUMNS` clause exposes the `city` partition as a column in the BigQuery
+  table schema.
+* The `require_hive_partition_filter=true` option means that all queries over this table require a
+  partition filter that can be used to eliminate partitions when reading data.
+
+External Hive table definition that matches the BigQuery table's schema:
+
+```sql
+CREATE EXTERNAL TABLE mydb.my_parquet_table (
+    id BIGINT,
+    name STRING
+    city STRING
+)
+STORED BY 'com.google.cloud.hive.bigquery.connector.BigQueryStorageHandler'
+TBLPROPERTIES (
+  'bq.table' = 'myproject.mydataset.my_parquet_table'
+)
+```
+
+Sample Hive query to read data:
+
+```sql
+SELECT *
+FROM mydb.my_parquet_table
+WHERE city = "Paris"
+```
+
+For more information, refer to the official BigQuery documentation on
```

Comment:
Also please add a reference to https://cloud.google.com/bigquery/docs/create-cloud-storage-table-biglake